src/pybel_tools/analysis/spia.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+
+"""This module contains the BEL exporter to SPIA format.
+(see: https://bioconductor.org/packages/release/bioc/html/SPIA.html)
+"""
+from itertools import product
+from typing import Dict
+
+import pandas as pd
+
+from pybel import BELGraph
+from pybel.constants import ASSOCIATION, INCREASES, DECREASES, DIRECTLY_DECREASES, DIRECTLY_INCREASES
+from pybel.dsl.node_classes import BaseAbundance, CentralDogma, Gene, ListAbundance, Rna, ProteinModification
+from pybel_tools.constants import KEGG_RELATIONS
+
+
+def get_matrix_index(graph: BELGraph) -> set:
+    """Return set of HGNC names from Proteins/Rnas/Genes/miRNA, nodes that can be used by SPIA.
+
+    :param graph: BELGraph
+    :return: node names
+    """
+    # TODO: Using HGNC Symbols for now
+    return {
+        node.name
+        for node in graph.nodes()
+        if issubclass(type(node), BaseAbundance) and node.namespace == 'HGNC'
+    }
+
+
+def build_matrices(nodes: set) -> Dict[str, pd.DataFrame]:
+    """Build adjacency matrices for each KEGG relationships.
+
+    :param nodes: nodes to be included in the matrix
+    :return: Dictionary of adjency matrix for each relationship
+    """
+    return {
+        relation: pd.DataFrame(0, index=nodes, columns=nodes)
+        for relation in KEGG_RELATIONS
+    }
+
+
+def update_matrix(matrix_dict, sub, obj, data):
+    """
+
+    :param matrix_dict:
+    :param sub:
+    :param obj:
+    :param data:
+    :return:
+    """
+    if sub.namespace == 'HGNC' or obj.namespace == 'HGNC':
+        subject_name = sub.name
+        object_name = obj.name
+
+        relation = data['relation']
+
+        if relation in {DIRECTLY_INCREASES, INCREASES}:
+
+            # If it has pmod check which one and add it to the corresponding matrix
+            if obj.variants and any(isinstance(variant, ProteinModification) for variant in obj.variants):
+
+                for variant in obj.variants:
+
+                    if variant['identifier']['name'] == "Ub":
+                        matrix_dict["activation_ubiquination"][subject_name][object_name] = 1
+
+                    elif variant['identifier']['name'] == "Ph":
```

Comment:
make generic functions for testing phosphorylation, especially if there are multiple valid possibilities (what about equivalent GO term?)

Reply:
Can you list all possibilities? Not sure whether I am aware of all the ones out there...

Reply:
You should use an `Enum` for all the possiblities @cthoyt